src/components/PopupWithForm.js

```diff
@@ -29,7 +29,8 @@ export default class PopupWithForm extends Popup {
     return inputsObject;
   }
 
-  _submitForm = () => {
+  _submitForm = (event) => {
+    event.preventDefault();
     const inputValues = this._getInputValues();
     this._handleFormSubmit(inputValues);
   };
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```